src/EditorFeatures/Core.Wpf/NavigateTo/NavigateToItemDisplay.cs

```diff
@@ -63,7 +64,7 @@ private ReadOnlyCollection<DescriptionItem> CreateDescriptionItems()
                 return new List<DescriptionItem>().AsReadOnly();
             }
 
-            var sourceText = document.GetTextSynchronously(CancellationToken.None);
+            var sourceText = document.GetTextSynchronously(document.Workspace.CurrentSolution, CancellationToken.None);
             var span = NavigateToUtilities.GetBoundedSpan(_searchResult.NavigableItem, sourceText);
```

Comment:
ick.  i never noticed this before.  This really violates a core principle of navto (that no work is done client side to avoid costly UI perf).  This is on the UI thread *and*  goes and potentially does IO.  Can you make issue that says that server should compute and add this info to items (ideally only if "show descriptions" is selected by user), and the client just has to present.  no need for GetTextSynchronously then.

Reply:
➡️ https://github.com/dotnet/roslyn/issues/68104